app/src/app/store/mapStore.ts

```diff
@@ -0,0 +1,102 @@
+import type { MapOptions } from "maplibre-gl";
+import { create } from "zustand";
+import type { ActiveTool, SpatialUnit } from "../constants/types";
+import { Zone, GEOID } from "../constants/types";
+export interface MapStore {
+  uuid: string | null;
+  setUuid: (uuid: string) => void;
+  mapOptions: MapOptions;
+  setMapOptions: (options: MapOptions) => void;
+  activeTool: ActiveTool;
+  setActiveTool: (tool: ActiveTool) => void;
+  spatialUnit: SpatialUnit;
+  setSpatialUnit: (unit: SpatialUnit) => void;
+  selectedZone: Zone;
+  setSelectedZone: (zone: Zone) => void;
+  zoneAssignments: Map<string, number>; // geoid -> zone
+  setZoneAssignments: (zone: Zone, geoids: Set<GEOID>) => void;
+  accumulatedGeoids: Set<string>;
+  brushSize: number;
+  setBrushSize: (size: number) => void;
+  isPainting: boolean;
+  setIsPainting: (isPainting: boolean) => void;
+}
+
+export const useMapStore = create<MapStore>((set) => ({
+  /**
+   * Unique identifier for the map instance
+   * @type {string | null}
+   */
+  uuid: null,
+  setUuid: (uuid: string) => set({ uuid: uuid }),
+  /**
+   * maplibre map instance options
+   * @type {MapOptions}
+   */
+  mapOptions: {
+    center: [-98.5795, 39.8283],
+    zoom: 3,
+    pitch: 0,
+    bearing: 0,
+    container: "",
+  },
+  setMapOptions: (options: MapOptions) => set({ mapOptions: options }),
+  /**
+   * Selected tool for the user to interact with the map
+   * @type {ActiveTool}
+   */
+  activeTool: "brush",
+  setActiveTool: (tool: ActiveTool) => set({ activeTool: tool }),
+  /**
+   * Spatial unit of geometry available to the user
+   * @type {SpatialUnit}
+   */
+  spatialUnit: "tract",
+  setSpatialUnit: (unit: SpatialUnit) => set({ spatialUnit: unit }),
+  /**
+   * Precinct zone currently being used to assign geoids via painting
+   * @type {Zone}
+   */
+  selectedZone: 1,
+  setSelectedZone: (zone: Zone) => set({ selectedZone: zone }),
+  /**
+   * Dictionary of geoid: zone assignments
+   * @type {Map<string, number>}
+   */
+  zoneAssignments: new Map(),
+  /**
+   * Set of geoids that have been assigned to a zone
+   * @type {Set<string>}
+   */
+  accumulatedGeoids: new Set<string>(),
+  /**
+   *
+   * @param zone - identifier for the zone assignment
+   * @param geoids - Set of geoids to assign to the zone; members of the set will be assigned to the zone
+   * @returns updated dict of geoid: zone assignments
+   */
+  setZoneAssignments: (zone: Zone, geoids: Set<GEOID>) =>
```

Comment:
Sadly for my pet theory it seems like maplibregl [doesn't support easily querying feature state](https://docs.mapbox.com/style-spec/reference/layers/#filter) for more than 1 id? So we'll have to have this temp store

Reply:
bummer, but copy that